src/forms/field_mapper.ts

```diff
@@ -0,0 +1,219 @@
+import {Tickets, UserField} from 'node-zendesk';
+import {AppField, AppCallValues} from 'mattermost-redux/types/apps';
+
+import {AppFieldTypes} from 'mattermost-redux/constants/apps';
+
+import {zdTypes, systemFields, fieldNames, mappedZDNames, makeFormOptions, makeOptions, formTextAreaMaxLength} from '../utils';
+
+// YET TO BE IMPLEMENTED
+// * custom types
+//   * DONE - multi-line
+//   *  X   - multi-select
+//   *  X   - decimal
+//   *  X   - numeric
+//   * DONE - checkbox
+//   *  X   - credit card
+//   *  X   - regex
+// * system types
+//   * DONE - subject
+//   * DONE - priority
+//   * DONE - tickettype
+//   * DONE - description
+//   *  X   - assignee
+
+// TODO: see what the limits are on text length for multi-line type
+
+// FIELD TYPES AVAILABLE IN plugin-apps/server/apps/field.go
+// FieldTypeText          = FieldType("text")
+// FieldTypeStaticSelect  = FieldType("static_select")
+// FieldTypeDynamicSelect = FieldType("dynamic_select")
+// FieldTypeBool          = FieldType("bool")
+// FieldTypeUser          = FieldType("user")
+// FieldTypeChannel       = FieldType("channel")
+
+interface Imapper {
+    mapFieldsToAppFields(fields: UserField[]): AppField[];
+    mapFormsToSelectField(forms: any): AppField;
+}
+
+export class FieldMapper implements Imapper {
+    private values: AppCallValues
+    private postMessage: string
+    constructor(values: AppCallValues, postMessage: string) {
```

Comment:
See if postmessage comes in values object.  If so it doesn’t need to be passed on.  Access directly from values object

Reply:
It comes in the context.  remove as and input to the FormFields constructor

Reply:
done.